owncloudApp/src/main/java/com/owncloud/android/presentation/settings/advanced/SettingsAdvancedViewModel.kt

```diff
@@ -34,4 +41,14 @@ class SettingsAdvancedViewModel(
     fun setShowHiddenFiles(hide: Boolean) {
         preferencesProvider.putBoolean(PREF_SHOW_HIDDEN_FILES, hide)
     }
+
+    fun scheduleDeleteLocalFiles(newValue: String) {
+        if (newValue == DeleteLocalFiles.NEVER.name) {
+            workManager.cancelAllWorkByTag(DELETE_FILES_OLDER_GIVEN_TIME_WORKER)
+        } else {
+            val timeSelected =
+                DeleteLocalFiles.valueOf(preferencesProvider.getString(PREFERENCE_DELETE_LOCAL_FILES, DeleteLocalFiles.NEVER.name)!!).toMilliseconds()
```

Comment:
The time selected does not deal with the time interval for workers to be triggered, but how old should be the local files to be removed by the worker. Workers will run every hour as basis, regardless the preference value. In case of a preference change, not sure if it is feasible to let the worker know the new parameter, or it is mandatory to remove and create a new one with the new value. But, both would work for me. I'll let you set the restriction ;)

Reply:
Done